build: vrrpd should only be compiled on Linux

Since vrrpd is only suppored on Linux, we should not try to compile it on other operating systems, which we do by default.

For instance, on OpenBSD this fails with:
```
  CC       vrrpd/vrrp.o
vrrpd/vrrp.c:1120:7: error: use of undeclared identifier 'SO_BINDTODEVICE'
                                         SO_BINDTODEVICE, r->mvl_ifp->name,
                                         ^
vrrpd/vrrp.c:1157:7: error: use of undeclared identifier 'SO_BINDTODEVICE'
                                         SO_BINDTODEVICE, r->vr->ifp->name,
                                         ^
vrrpd/vrrp.c:1213:19: error: variable has incomplete type 'struct ip_mreqn'
                struct ip_mreqn mreqn = {};
                                ^
vrrpd/vrrp.c:1213:10: note: forward declaration of 'struct ip_mreqn'
                struct ip_mreqn mreqn = {};
                       ^
vrrpd/vrrp.c:1267:7: error: use of undeclared identifier 'SO_BINDTODEVICE'
                                         SO_BINDTODEVICE, r->vr->ifp->name,
                                         ^
4 errors generated.
```
Some checks will have to be added to configure.ac to prevent this.

--- a/configure.ac
+++ b/configure.ac
@@ -1791,6 +1791,22 @@ if test "$enable_rpki" = "yes"; then
   )
 fi
 
+dnl -------------------------------------
+dnl VRRP is only supported on linux
+dnl -------------------------------------
+if test "$enable_vrrpd" != "no"; then
+AC_MSG_CHECKING([for VRRP OS support])
+case "$host_os" in
+  linux*)
+    AC_MSG_RESULT([yes])
+    ;;
+  *)
+    AC_MSG_RESULT([no])
+    enable_vrrpd="no"
+    ;;
+esac
+fi
+
 dnl ------------------------------------------
 dnl Check whether rtrlib was build with ssh support
 dnl ------------------------------------------

